### PR TITLE
Fix ESP32, compilation issues and bug fixes

### PR DESCRIPTION
**Pull request by aburt2**
# Fix compilation issues and minor errors
- fixed syntax issue not caught by compiler when selecting serial monitor (accidently used single equal signs instead of double equal signs)
- fix linker issues when compiling for ESP32 #6 
- fixed logic for puara.getIPX_ready() to return false when osc address is either _0.0.0.0_ or _""_ #7 
- fix oscPORT2 not being displayed in the webpage #8

### CODE REVIEW

**Comment by aburt2** [comment]
@edumeneses I'll use this pull request to also fix the ESP32 compilation errors due to the previous pull request, I'll change this to draft and ping you once complete

**Comment by aburt2** [comment]
works on the ESP32S3, still need to test on ESP32 device but it compiles for ESP32 target

**Comment by edumeneses** [comment]
Nice.

What is the current workflow for using puara-module at IDMIL? Are we building on PlatformIO?

File: puara.cpp [hunk 1]
```diff
@@ -1497,17 +1501,17 @@ std::string Puara::getLocalPORTStr() {
 }
 
 bool Puara::IP1_ready() {
-    if (oscIP1 != "0.0.0.0" || oscIP1 != "") {
+    if ((oscIP1 != "0.0.0.0") && (oscIP1 != "")) {
         return true;
     } else {
         return false;
     }
 }
```
Comment (edumeneses) [inline puara.cpp:1509]: Another possibility to make it more readable IMO is 

```suggestion
    if ((oscIP1 == "0.0.0.0") || (oscIP1 == "")) {
        return false;
    } else {
        return true;
    }
}
```

We can also use the ternary operator (I think makes it more readable in this case):

`return (oscIP1 == "0.0.0.0" || oscIP1 == "") ? false : true;`

File: puara.cpp [hunk 2]
```diff
@@ -1497,17 +1501,17 @@ std::string Puara::getLocalPORTStr() {
...
 
 bool Puara::IP2_ready() {
-    if (oscIP2 != "0.0.0.0" || oscIP2 != "") {
+    if ((oscIP2 != "0.0.0.0") && (oscIP2 != "")) {
         return true;
     } else {
         return false;
     }
```
Comment (edumeneses) [inline puara.cpp:1516]: Same suggestion as above:

`return (oscIP2 == "0.0.0.0" || oscIP2 == "") ? false : true;`

File: puara.cpp (hunk 1 above)
Comment (aburt2) [reply puara.cpp]: definitely, I'll make the change

**Review by edumeneses — COMMENTED** [review]
Looks good to me so far

**Comment by edumeneses** [comment]
The (hopefully) last bug is related to the `sta_event_handler` and passing an `int` instead of an `int32_t` for the event_id. I will try to fix it.

**Comment by edumeneses** [comment]
I will merge this to finish the fix

**Comment by edumeneses** [comment]
Merging as is and finishing the fixing on my end. Thanks @aburt2 for the work!

**Pull request merged**